### PR TITLE
feat: [FFM-1223]: Flag settings panel

### PR DESCRIPTION
**Pull request by knagurski**
This PR adds the Flag Settings panel of the new Target Groups page. ONLY rules created as `segmentMatch` will currently work with this, pending a backend migration to transform `TargetMap` rules.

|Before|After|
|------|------|
|Before|After|

Refs [FFM-1223](https://harness.atlassian.net/browse/FFM-1223)

### Additions
- added `FlagDetailsCell` component and tests
- added `FlagSettingsForm` component, styling and tests
- added `FlagSettingsFormButtons` component, styling and tests
- added `FlagSettingsPanel` component and tests
- added `VariationsCell` component and tests
- added `flagSettingsInstructions` utility file and tests
- added `getFlagSettingsInstructions` utility function and tests
- added `getRemovedFlagIdentifiers` utility function and tests
- added `getUpdatedFlags` utility function and tests
- added `getRemoveFlagsInstruction` utility function and tests
- added `getUpdateFlagsInstruction` utility function and tests
- added additional strings

### Changes
- moved teepee image to shared location
- updated `FeatureFlagsPage` to point at new location of teepee image
- ensured target group loading spinner centres

### CODE REVIEW

**Comment by knagurski** [comment]
trigger cypress

File: src/modules/75-cf/pages/target-group-detail/components/FlagSettingsPanel/VariationsCell.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  flag: TargetGroupFlag
+  fieldPrefix: string
+}
+
+const VariationsCell: FC<VariationsCellProps> = ({ flag, fieldPrefix }) => {
+  const { getString } = useStrings()
+
+  const flagItems = useMemo<SelectOption[]>(
+    () =>
+      flag.flag.variations.map<SelectOption>(({ identifier, name }) => ({
```
Comment (crystal-fong) [inline src/modules/75-cf/pages/target-group-detail/components/FlagSettingsPanel/VariationsCell.tsx:23]: is it worth destructuring this out to better readability?

Comment (knagurski) [reply src/modules/75-cf/pages/target-group-detail/components/FlagSettingsPanel/VariationsCell.tsx:23]: Good call. `flag.flag.variations.map` doesn't exactly roll off the tongue!

Comment (knagurski) [reply src/modules/75-cf/pages/target-group-detail/components/FlagSettingsPanel/VariationsCell.tsx]: More betterer now?

Comment (crystal-fong) [reply src/modules/75-cf/pages/target-group-detail/components/FlagSettingsPanel/VariationsCell.tsx]: 🥇 betterer!

File: src/modules/75-cf/pages/target-group-detail/components/FlagSettingsPanel/FlagSettingsForm.tsx [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+            )
+
+        return (
+          <>
+            <Page.SubHeader className={css.toolbar}>
+              <ExpandingSearchInput alwaysExpanded name="flag-search" onChange={text => setSearchTerm(text)} />
+            </Page.SubHeader>
+
+            <FormikForm className={css.formLayout} disabled={submitting}>
+              {(!searchTerm || filteredFlags.length > 0) && (
```
Comment (chiragp1212) [inline src/modules/75-cf/pages/target-group-detail/components/FlagSettingsPanel/FlagSettingsForm.tsx:148]: these render conditions can be memoised 

const showTable = useMemo(() => {return !searchTerm || !isEmpty(filteredFlags) }, [searchTerm, filteredFlags])

**Review by chiragp1212 — APPROVED** [review]

File: src/modules/75-cf/pages/target-group-detail/components/FlagSettingsPanel/FlagSettingsForm.tsx (hunk 2 above)
Comment (knagurski) [reply src/modules/75-cf/pages/target-group-detail/components/FlagSettingsPanel/FlagSettingsForm.tsx:148]: Unfortunately they can't. A hook can only be used within the body of a component, the data (`filteredFlags`) is only available within this scope 😞

Comment (chiragp1212) [reply src/modules/75-cf/pages/target-group-detail/components/FlagSettingsPanel/FlagSettingsForm.tsx:148]: I see. But they can be extracted in a functional component and then used that way.

<FormikForm className={css.formLayout} disabled={submitting}>
   {() => <FormContent {..pass the props} /> )}
</FormikForm>

Not a blocking comment though

Comment (knagurski) [reply src/modules/75-cf/pages/target-group-detail/components/FlagSettingsPanel/FlagSettingsForm.tsx:148]: Good point. My next ticket is in this area as well, so I'll refactor as part of that. Thanks for the suggestions 👍🏻

**Review by chiragp1212 — APPROVED** [review]

**Review by crystal-fong — APPROVED** [review]

**Pull request merged**